RIOT-OS/RIOT
# rename cpu_init() to soc_init(), don't call it from board_init()

### Contribution description

There is a inconsistency in RIOT: Some platforms (arm7, esp*, native) will call both `board_init()` and `cpu_init()` from` `cpu/` whereas other platforms expect boards to call `cpu_init()` from `board_init()`.

The former approach is cleaner and feels more 'natural'. So convert the other platforms to also follow this pattern.

Since doing so would break every external board, this also renames `cpu_init()` to `soc_init()` so we can introduce a no-op `cpu_init()` for compatibility.

`soc_init()` is now always called before `board_init()`.


### Testing procedure

Everything should still work as before.
I did not remove the calls to `cpu_init()` from boards yet, as this simulates not updated external baords.
Once a Murdock run was successful, I will push a follow-up patch to remove `cpu_init()` calls from `boards/`. 


### Issues/PRs references
